lib/honeybadger/config/ruby.rb

```diff
@@ -86,6 +86,18 @@ def backtrace_filter
         get(:backtrace_filter)
       end
 
+      def before_notify(action = nil, &block)
+        hash[:before_notify] ||= []
+
+        if action
+          hash[:before_notify] << action
```

Comment:
Should we check that `action` responds to `#call`? Another check could be that it has an `arity` of one. This might look like:

```ruby
if action.respond_to?(:call) && action.arity == 1
  hash[:before_notify]
else
  raise SomeErrorClass, "#{action} is not a before notify hook (responds to call and accepts one arg)"
end
```

Reply:
This is a good idea :+1:
However, instead of raising we should probably log an error and not cause the whole app to crash.